bot.rb

```diff
@@ -65,7 +66,17 @@
   command = Commands.const_get(c)
   command.is_a?(Class) ? command : nil
 end.compact.each do |command|
-  BOT.command(command.name, command.attributes, &command.method(:command))
+  method = command.method(:command).to_proc
+  middleware = [
+    method(:CheckAbove)
+  ]
+  method.define_singleton_method(:call) do |event, *args|
+    middleware.each do |m|
+      *args = m.call(event, *args)
```

Comment:
Strictly speaking, it's a code smell to re-assign arguments. Better to move them to their own variable.

You also don't need the `*` in your assignment.

```ruby
transformed_args = args.dup
middleware.each do |m|
  transformed_args = m.call(event, *transformed_args)
end
super(event, *transformed_args)